shlib/server.sh

```diff
@@ -46,24 +46,28 @@ prepare_server() {
   if test "$MONGODB_VERSION" = latest; then
     # Test on the most recent published 4.3 release.
     # https://jira.mongodb.org/browse/RUBY-1724
-    echo 'Using "latest" server is not currently implemented' 1>&2
-    exit 1
+
+    . $PROJECT_DIRECTORY/.mod/drivers-evergreen-tools/.evergreen/download-mongodb.sh
+
+    get_distro
+    get_mongodb_download_url_for "$DISTRO" "latest"
+    prepare_server_from_url $MONGODB_DOWNLOAD_URL
   else
     download_version="$MONGODB_VERSION"
+    url=`$(dirname $0)/get-mongodb-download-url $download_version $arch`
+    prepare_server_from_url $url
   fi
 
-  url=`$(dirname $0)/get-mongodb-download-url $download_version $arch`
-
-  prepare_server_from_url $url
 }
 
 prepare_server_from_url() {
   url=$1
 
-  mongodb_dir="$MONGO_ORCHESTRATION_HOME"/mdb
+  dirname=`basename $url |sed -e s/.tgz//`
+  mongodb_dir="$MONGO_ORCHESTRATION_HOME"/mdb/"$dirname"
   mkdir -p "$mongodb_dir"
-  curl --retry 3 $url |tar xz -C "$mongodb_dir" -f -
-  BINDIR="$mongodb_dir"/`basename $url |sed -e s/.tgz//`/bin
+  curl --retry 3 $url | tar xz -C "$mongodb_dir" --strip-components 1 -f -
```

Comment:
Does this work for non-latest too?

Reply:
I believe it does... However I am running into errors, although I think they might be unrelated... I will let you know when I have green CI

Reply:
Yes it does work for non-latest. What I'm doing here is I'm setting the directory that it's going to be stored in (derived from the url), instead of just assuming that the directory name that we derive from the url is automatically going to match the one unwrapped from the TAR file. It almost always does, except when using the `-latest` URL.

Reply:
CI looks to be green! https://github.com/mongodb/mongo-ruby-driver/pull/2473